src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -157,28 +172,48 @@ private static string BuildTopLevelSelect(SelectExpandClause selectExpandClause)
         /// </summary>
         /// <param name="selectedItem">the select item to translate</param>
         /// <returns>the string representation of this select item, or null if the select item is an expandedNavPropSelectItem</returns>
-        private static string GetSelectString(SelectItem selectedItem)
+        private static IList<string> GetSelectString(SelectItem selectedItem)
         {
             WildcardSelectItem wildcardSelect = selectedItem as WildcardSelectItem;
             NamespaceQualifiedWildcardSelectItem namespaceQualifiedWildcard = selectedItem as NamespaceQualifiedWildcardSelectItem;

```

Comment:
can move these two cases into "GetCurrentLevelSelectList" method.

Reply:
This method can only process the PathSelectItem

---
In reply to: [356798601](https://github.com/OData/odata.net/pull/1607#discussion_r356798601) [](ancestors = 356798601)

Reply:
Resolved

Reply:
If you resolve one comment, would you please mark that comment as "#Resolved"

---
In reply to: [356881113](https://github.com/OData/odata.net/pull/1607#discussion_r356881113) [](ancestors = 356881113)

Reply:
#Resolved